python/lsst/cp/pipe/ptc.py

```diff
@@ -446,12 +413,42 @@ def runDataRef(self, dataRef, visitPairs):
 
             butler.put(linearizer, datasetType='Linearizer', dataId={'detector': detNum,
                        'detectorName': detName, 'calibDate': calibDate})
-
-        self.log.info(f"Writing PTC data to {dataRef.getUri(write=True)}")
+        rerunDir = list(dataRef.getButler().storage.repositoryCfgs.keys())[0]
+        self.log.info(f"Writing PTC data to {rerunDir} directory.")
         dataRef.put(datasetPtc, datasetType="photonTransferCurveDataset")
 
         return pipeBase.Struct(exitStatus=0)
 
+    def makePairs(self, dataRefList):
+        """Produce a list of flat pairs indexed by exposure time.
+
+        Parameters
+        ----------
+        dataReflist : `list` [`lsst.daf.peristence.ButlerDataRef`]
```

Comment:
Is there a plan for making this gen3 compatible?  (also typo in the class name).

Reply:
This is step one of that conversion.  https://jira.lsstcorp.org/browse/DM-21786 is the umbrella ticket for this.

Reply:
Thanks for pointing out the typo. Regarding gen3 compatibility, I'm not sure about what you are referring to. Are you talking about the "makePairs" function in particular? Or the PTC task in general? If the latter, yes, this ticket is part of a larger ticket (DM-21786) to convert the PTC task to gen3.

Reply:
I was just noting that this is brand new code that will have to be rewritten on a November timeline.

Reply:
The idea is that after DM-21786 the PTC task will be fully converted to gen3.